### PR TITLE
MRD-916 Endpoint for recommendations tab

### PR DESCRIPTION
**Pull request by jackmaloney**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/domain/makerecalldecisions/recommendation/RecommendationTabStatus.kt [hunk 1]
```diff
@@ -0,0 +1,5 @@
+package uk.gov.justice.digital.hmpps.makerecalldecisionapi.domain.makerecalldecisions.recommendation
+
+enum class RecommendationTabStatus {
+  CONSIDERING_RECALL, RECOMMENDATION_STARTED, MAKING_DECISION_TO_RECALL, MAKING_DECISION_NOT_TO_RECALL, DECIDED_TO_RECALL, DECIDED_NOT_TO_RECALL, UNKNOWN
```
Comment (jonnywyatt) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/domain/makerecalldecisions/recommendation/RecommendationTabStatus.kt:4]: What is 'UNKNOWN' for?

File: src/main/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/domain/makerecalldecisions/recommendation/RecommendationTab.kt [hunk 2]
```diff
@@ -0,0 +1,8 @@
+package uk.gov.justice.digital.hmpps.makerecalldecisionapi.domain.makerecalldecisions.recommendation
+
+data class RecommendationTab(
```
Comment (jonnywyatt) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/domain/makerecalldecisions/recommendation/RecommendationTab.kt:3]: It's minor, but the API doesn't need to know how it's presented, is `RecommendationsListItem` better?

**Review by bills7281 — DISMISSED** [review]

File: src/main/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/domain/makerecalldecisions/recommendation/RecommendationTabStatus.kt (hunk 1 above)
Comment (jackmaloney) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/domain/makerecalldecisions/recommendation/RecommendationTabStatus.kt:4]: Should never happen, but I had to default it to something in the event someone in the future overwrites part of a case and we can't determine the status for whatever reason. Basically we can't test for this so wouldn't worry too much.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/domain/makerecalldecisions/recommendation/RecommendationTab.kt (hunk 2 above)
Comment (jackmaloney) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/domain/makerecalldecisions/recommendation/RecommendationTab.kt]: Updated - was in 2 minds what to call this. We have alot of similarly named classes.

**Review by jonnywyatt — APPROVED** [review]

**Review by bills7281 — APPROVED** [review]

**Pull request merged**